controllers/managedClusterResources.go

```diff
@@ -250,3 +260,306 @@ func (r *ClusterGroupUpgradeReconciler) renderYamlTemplate(
+		return UnforeseenCondition, err
+	}
+	if !present {
+		return NoJobView, nil
+	}
+	viewConditions, exists, err := unstructured.NestedSlice(
+		jobView.Object, jobsInitialStatus...)
+	if !exists {
+		return UnforeseenCondition, fmt.Errorf(
+			"[getActiveConditions] no ManagedClusterView conditions found")
+	}
+	if err != nil {
+		return UnforeseenCondition, err
+	}
+	if !r.checkViewProcessing(viewConditions) {
+		return NoJobFoundOnSpoke, nil
+	}
+	return r.getJobStatus(jobView)
+}
+
+// checkDependenciesViews check all precache job dependencies views
+//		have been deployed
+// returns: available (bool)
+//			error
+func (r *ClusterGroupUpgradeReconciler) checkDependenciesViews(
+	ctx context.Context, cluster string) (bool, error) {
+
+	for _, item := range precacheDependenciesViewTemplates {
+		_, available, err := r.getView(
+			ctx, item.resourceName, cluster)
+		if err != nil {
+			return false, err
+		}
+		if !available {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// checkDependencies check all precache job dependencies are available
+// returns: 	available (bool)
+//				error
+func (r *ClusterGroupUpgradeReconciler) checkDependencies(
+	ctx context.Context, cluster string) (bool, error) {
+
+	for _, item := range precacheDependenciesViewTemplates {
+		view, available, err := r.getView(
+			ctx, item.resourceName, cluster)
+		if err != nil {
+			return false, err
+		}
+		if !available {
+			return false, nil
+		}
+		viewConditions, exists, err := unstructured.NestedSlice(
+			view.Object, jobsInitialStatus...)
+		if !exists {
+			return false, fmt.Errorf(
+				"[getPreparingConditions] no ManagedClusterView conditions found")
+		}
+		if err != nil {
+			return false, err
+		}
+		present := r.checkViewProcessing(viewConditions)
+		if !present {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// getPrecacheJobTemplateData initializes template data for the job creation
+// returns: 	*templateData
+//				error
+func (r *ClusterGroupUpgradeReconciler) getPrecacheJobTemplateData(
+	ctx context.Context,
+	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string) (
+	*templateData, error) {
+
+	rv := new(templateData)
+
+	rv.Cluster = clusterName
+	rv.JobTimeout = uint64(
+		clusterGroupUpgrade.Spec.RemediationStrategy.Timeout) * 60
+	image, err := r.getPrecacheimagePullSpec(ctx, clusterGroupUpgrade)
+	if err != nil {
+		return rv, err
+	}
+	rv.WorkloadImage = image
+	return rv, nil
+}
+
+// deployPrecachingWorkload deploys precaching workload on the spoke
+//          using a set of templated manifests
+// returns: error
+func (r *ClusterGroupUpgradeReconciler) deployWorkload(
+	ctx context.Context,
+	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade,
+	cluster, workloadType, mcvTemplate string, template []resourceTemplate) error {
+
+	// the idea is to reuse the deployworkload function for backup job, so using switch can different condition different jobs
+	var spec *templateData
+	switch workloadType {
```

Comment:
golangci-lint test is failing here, suggesting an if-check rather than a single-case switch

Reply:
i will add default for now, the idea was, with the next PR there will be another workloadtype "backup"

Reply:
fixed and squashed to previous commit.